Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -4732,29 +4809,24 @@ private void DoGetRecordings()
                             string ControlAction = PLR.GetValueString();
                             string Type = PLR.GetValueString();
 
-                            if (ControlAction == "SetValue")
-                                ControlAction = "SendKeys";
-                            if (ControlAction == "Click" && (Type.ToLower() == "a" || Type.ToLower() == "submit"))
+                            //if (ControlAction == "SetValue")
+                            //    ControlAction = "SendKeys";
+                            if (ControlAction.ToLower() == "click" && (Type.ToLower() == "a" || Type.ToLower() == "submit"))
                                 Thread.Sleep(2000);
 
-                            //TODO: switch to use ActUIElememnt
-                            ActGenElement act = new ActGenElement();
-                            act.Description = GetDescription(ControlAction, LocateValue, ElemValue, Type);
-                            act.LocateBy = GetLocateBy(LocateBy);
-                            act.LocateValue = LocateValue;
-                            act.GenElementAction = GetElemAction(ControlAction);
-                            act.Value = ElemValue;
-                            this.BusinessFlow.AddAct(act);
+                            ActUIElement actUI = new ActUIElement();
+                            actUI.Description = GetDescription(ControlAction, LocateValue, ElemValue, Type);
+                            actUI.ElementLocateBy = GetLocateBy(LocateBy);
+                            actUI.ElementLocateValue = LocateValue;
+                            actUI.ElementType = GetUIElementTypeByPayLoadType(Type);
+                            if (Enum.IsDefined(typeof(ActUIElement.eElementAction), ControlAction))
```

Comment:
JS "Type" also need to be the string of eElementType already so only need to convert it from String to Enum instead using the GetUIElementTypeByPayLoadType() method